src/rust/engine/process_execution/src/local.rs

```diff
@@ -288,7 +288,19 @@ impl CapturedWorkdir for CommandRunner {
     _context: Context,
     exclusive_spawn: bool,
   ) -> Result<BoxStream<'c, Result<ChildOutput, String>>, String> {
-    let mut command = HermeticCommand::new(&req.argv[0]);
+    // TODO(#11406): Go back to using relative paths, rather than absolutifying them, once Rust
+    //  fixes https://github.com/rust-lang/rust/issues/80819 for macOS.
+    let argv0 = match RelativePath::new(&req.argv[0]) {
+      Ok(rel_path) => {
+        if let Some(ref working_directory) = req.working_directory {
+          workdir_path.join(working_directory).join(rel_path)
```

Comment:
This was a little confusing to me to reason about, for some reason. I'd appreciate an ack my thinking is correct.

Note that we right below will set the cwd like this:

https://github.com/pantsbuild/pants/blob/8d4182a7768d85556061e7b167f3c1554ecbe6bf/src/rust/engine/process_execution/src/local.rs#L294-L298

Reply:
The logic looks correct, but it seems like it would help to lift that lower calculation out and re-use it for both argv0 absolutification and curent_dir setting.